client/packages/common/src/ui/layout/tables/context/TableContext.tsx

```diff
@@ -142,7 +142,10 @@ export const createTableStore = () =>
 
         // Set new styles for the ids passed.
         ids.forEach(id => {
-          rowState[id] = getRowState(state, id, { style });
+          const currentRowStyle = rowState?.[id]?.style ?? {};
+          rowState[id] = getRowState(state, id, {
+            style: { ...currentRowStyle, ...style } as AppSxProp,
+          });
```

Comment:
This allows sequential "setRowStyles" calls to update the row style rather than replace it.

Reply:
:nit in which case the 'set' is misleading and may well cause issues later. Can you instead add an 'updateRowStyle' method? Also with a change like this - are all current usages happy about an update rather than a set?